--- a/net/minecraft/world/item/alchemy/Potion.java
+++ b/net/minecraft/world/item/alchemy/Potion.java
@@ -16,7 +16,7 @@
 
    public static Holder<Potion> m_43489_(String p_43490_) {
       ResourceLocation resourcelocation = ResourceLocation.m_135820_(p_43490_);
-      return resourcelocation == null ? Potions.f_43598_ : BuiltInRegistries.f_256980_.m_313673_(resourcelocation).map(Function.identity()).orElse(Potions.f_43598_);
+      return resourcelocation == null ? Potions.f_43598_ : BuiltInRegistries.f_256980_.m_313673_(resourcelocation).<Holder<Potion>>map(Function.identity()).orElse(Potions.f_43598_);
    }
 
    public Potion(MobEffectInstance... p_43487_) {
