--- a/net/minecraft/resources/HolderSetCodec.java
+++ b/net/minecraft/resources/HolderSetCodec.java
@@ -66,7 +66,7 @@
    }
 
    private static <E> DataResult<HolderSet<E>> m_330630_(HolderGetter<E> p_330991_, TagKey<E> p_330977_) {
-      return p_330991_.m_255050_(p_330977_).map(DataResult::success).orElseGet(() -> {
+      return p_330991_.m_255050_(p_330977_).<DataResult<HolderSet<E>>>map(DataResult::success).orElseGet(() -> {
          return DataResult.error(() -> {
             return "Missing tag: '" + p_330977_.f_203868_() + "' in '" + p_330977_.f_203867_().m_135782_() + "'";
          });
