--- a/net/minecraft/world/item/component/DebugStickState.java
+++ b/net/minecraft/world/item/component/DebugStickState.java
@@ -13,7 +13,7 @@
 public record DebugStickState(Map<Holder<Block>, Property<?>> f_326030_) {
    public static final DebugStickState f_325447_ = new DebugStickState(Map.of());
    public static final Codec<DebugStickState> f_325471_ = ExtraCodecs.m_327848_(BuiltInRegistries.f_256975_.m_206110_(), (p_329285_) -> {
-      return Codec.STRING.comapFlatMap((p_328593_) -> {
+      return Codec.STRING.<Property<?>>comapFlatMap((p_328593_) -> {
          Property<?> property = p_329285_.m_203334_().m_49965_().m_61081_(p_328593_);
          return property != null ? DataResult.success(property) : DataResult.error(() -> {
             return "No property on " + p_329285_.m_313865_() + " with name: " + p_328593_;
