--- a/net/minecraft/core/registries/BuiltInRegistries.java
+++ b/net/minecraft/core/registries/BuiltInRegistries.java
@@ -320,7 +320,7 @@
       f_256751_.put(resourcelocation, () -> {
          return p_259210_.m_257957_(p_260327_);
       });
-      f_256935_.m_255290_(p_259230_, p_260327_, RegistrationInfo.f_323051_);
+      f_256935_.m_255290_((ResourceKey)p_259230_, p_260327_, RegistrationInfo.f_323051_);
       return p_260327_;
    }
 
